### PR TITLE
Features/body classification data module

### PR DESCRIPTION
**Pull request by jasonadriann**
# Description

Adding lightning data module and other necessary components for classification task using VinDr-BodyPartXR dataset.

Source: 
* [Lightning Data Module](https://lightning.ai/docs/pytorch/stable/data/datamodule.html)
* [VinDr-BodyPartXR](https://vindr.ai/datasets/bodypartxr) dataset

New Features:
* Add VinDr-BodyPartXR dataset parser based on [ActiveLoop](https://app.activeloop.ai/) dataset format
* Create LightningDataModule to read from [ActiveLoop](https://app.activeloop.ai/) dataset for VinDr-BodyPartXR
* Create testing functions and scenarios

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

The pr has been tested using [pytest](https://docs.pytest.org/en/7.4.x/) and dummy dataset that are being built with [tmp_path_factory](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html)

- [x] Test for ActiveLoop training, validation, testing dataset parser
- [x] Test for VinDr-BodyPartXR DataLightningModule

**Test Configuration**:

- Firmware version:
- Hardware: IdeaPad Gaming 3 + RTX3050

# Checklist:

- [x] Did you make sure title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by haritsahm** [comment]
Please add `deeplake` to `requirements.txt` others section

**Comment by jasonadriann** [comment]
Already adding `deeplake` to `requirements.txt` in the latest commit

**Review by haritsahm — COMMENTED** [review]
Please change and update the following sections

File: src/data/components/bodypartxr_parser.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Parameters
+    ----------
+    dataset_dir : str
+        Activeloop dataset directory
+    transform : Optional[torchvision.transforms.Compose], optional
+        Data augmentation pipeline, by default None
+    """
+
+    def __init__(self, transform: Optional[torchvision.transforms.Compose] = None, dataset_dir: Optional[str] = None):
+        # for split in ['train', 'val', 'test']:
```
[the remark this answers is not in the document]
Comment (jasonadriann) [reply src/data/components/bodypartxr_parser.py]: Already removed

File: src/data/components/bodypartxr_parser.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import deeplake
+import numpy as np
+import torch
+import torchvision
+from torch.utils.data import Dataset
+
+
+class VinDrBodyPartXRDataset(Dataset):
+    """VinDr Body Part XR Parser.
+
```
[the remark this answers is not in the document]
Comment (jasonadriann) [reply src/data/components/bodypartxr_parser.py:18]: Adding several explanations, please kindly re-review it

**Comment by haritsahm** [comment]
Please merge `dummy_bodypartxr_dataset.py` with `dummy_dataset.py`. Rename the function and update the test functions

**Comment by haritsahm** [comment]
Solved issue #13

**Pull request merged**